### DISCUSSION TITLE
How to Update Existing Components on Canvas When Component Definition Changes?

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#6016] [Q&A]
How can I ensure that existing components on the canvas are updated to reflect changes made to their component definitions after they have been dropped onto the canvas in GrapesJS?

@artf

**Comment by artf** [comment]
you should remove and re-create them

**Reply by RakulAgn** [reply]
While the Development phase its okie but think of it what if is there is a future requirement due to that we want to update the component definition if we update that the old templates which are created by users for those templates the updated Definition will not work right ? we can't tell 1000 of users to remove and add those components i think we have need some solution for this use case

**Reply by artf** [reply]
I was referring to a case where you're updating the component definition and want the current components already created to be updated in the canvas (btw it could be done via API, not manually).

If you're talking about loading a project containing `{ type: 'my-custom-component' }` and you updated its definition in code, well the component in the canvas will be loaded with the new update, you don't need to do anything.

**Reply by RakulAgn** [reply]
Ah.. So Doing Through Api is Something like migrating the Component Definition Right?
